modules/customize-certificates-api-update-named.adoc

```diff
@@ -0,0 +1,103 @@
+// Module included in the following assemblies:
+//
+// * security/certificates/api-server.adoc
+
+:_content-type: PROCEDURE
+[id="customize-certificates-api-update-named_{context}"]
+= Updating an API server named certificate
+
+You can update the API server certificate to a new version by creating a new secret and updating the API server.
+
+.Procedure
+
+. Log in to the API as the `kubeadmin` user.
++
+[source,terminal]
+----
+$ oc login -u kubeadmin -p <password> https://FQDN:6443
+----
+
+. Create a secret with a new name in the `openshift-config` namespace that contains the certificate chain and private key.
++
+[source,terminal]
+----
+$ oc create secret tls <secret> \//<1>
+     --cert=</path/to/cert.crt> \//<2>
+     --key=</path/to/cert.key> \//<3>
+     -n openshift-config
+----
+<1> `<secret>` is the name of the secret that will contain the certificate chain and private key.
+<2> `</path/to/cert.crt>` is the path to the certificate chain on your local file system.
+<3> `</path/to/cert.key>` is the path to the private key associated with this certificate.
++
+[NOTE]
+====
+Do not overwrite or delete the previous secret. Create a secret with a new name, for example `api-certs-<date>`. If you have to roll back, you can switch back to using the original secret.
+====
+
+. Update the API server to reference the secret with new API certificate.
++
+[source,terminal]
+----
+$ oc patch apiserver cluster \
+     --type=merge -p \
+     '{"spec":{"servingCerts": {"namedCertificates":
+     [{"names": ["<FQDN>"], //<1>
+     "servingCertificate": {"name": "<secret>"}}]}}}' <2>
+----
+<1> Replace `<FQDN>` with the FQDN that the API server should provide the certificate for.
+<2> Replace `<secret>` with the name used for the secret in the previous step.
+
+. Examine the `apiserver/cluster` object and confirm the secret is now
+referenced.
++
+[source,terminal]
+----
+$ oc get apiserver cluster -o yaml
+----
++
+.Example output
+[source,terminal]
+----
+...
+spec:
+  servingCerts:
+    namedCertificates:
+    - names:
+      - <FQDN>
+      servingCertificate:
+        name: <secret>
+...
+----
+
+. Verify that the new certificate is in place.
++
+[source,terminal]
+----
+$ curl -kv https://api.<cluster_name>.<base_domain> 2>&1 | grep -A5 -i "server certificate"
+----
++
+.Example output
+[source,terminal]
+----
+* Server certificate:
+* 	subject: CN=api.<cluster_name>.<base_domain>
+* 	start date: Aug 07 03:00:01 2023 GMT
+* 	expire date: Aug 06 03:00:01 2025 GMT
+* 	common name: api.<cluster_name>.<base_domain>
+* 	issuer: CN=Company Root CA Issuer
+----
+
+. Verify that `kube-apiserver` operator is healthy and not degraded. 
++
+[source,terminal]
+----
+$ oc get clusteroperators kube-apiserver
+----
++
+.Example output
+[source,terminal,subs="attributes+"]
+----
+NAME             VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE
```

Comment:
missed one `MESSAGE` here, Currently, the output looks like this.
```
NAME                                       VERSION                              AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
    authentication                             4.15.0-0.nightly-2023-12-09-012410   True        False         False      6m31s   
    baremetal                                  4.15.0-0.nightly-2023-12-09-012410   True        False         False      4h9m    
    cloud-controller-manager                   4.15.0-0.nightly-2023-12-09-012410   True        False         False      4h11m   
    cloud-credential                           4.15.0-0.nightly-2023-12-09-012410   True        False         False      4h11m   
    cluster-autoscaler                         4.15.0-0.nightly-2023-12-09-012410   True        False         False      4h9m    
    config-operator                            4.15.0-0.nightly-2023-12-09-012410   True        False         False      4h9m    
    console                                    4.15.0-0.nightly-2023-12-09-012410   True        False         False      44m     
    control-plane-machine-set                  4.15.0-0.nightly-2023-12-09-012410   True        False         False      57s     
    csi-snapshot-controller                    4.15.0-0.nightly-2023-12-09-012410   True        False         False      4h9m    
    dns                                        4.15.0-0.nightly-2023-12-09-012410   True        False         False      4h8m    
    etcd                                       4.15.0-0.nightly-2023-12-09-012410   True        False         False      4h7m    
    image-registry                             4.15.0-0.nightly-2023-12-09-012410   True        False         False      173m    
    ingress                                    4.15.0-0.nightly-2023-12-09-012410   True        False         False      173m    
    insights                                   4.15.0-0.nightly-2023-12-09-012410   True        False         False      4h3m    
    kube-apiserver                             4.15.0-0.nightly-2023-12-09-012410   True        False         False      4h6m    
...
```

Reply:
What a message is missed? Sorry, I don't understand the comment. Could you please elaborate? I did the command `oc get clusteroperators kube-apiserver` on my `4.14` and it works as expected.

Also on my `4.15` it is as expected:

~~~
# oc get clusteroperators kube-apiserver

NAME             VERSION                              AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
kube-apiserver   4.15.0-0.nightly-2024-01-25-051548   True        False         False      9m24s   
~~~

Reply:
Sorry, made you misunderstand, currently,  4.15 has one more column `MESSAGE` in header of the output, before this release, there is no this column.
4.15
```
NAME             VERSION                              AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
```